Set up project to be uploadable to Pypi

Guptacos opened the pull request:
- Update gitignore for setuptools builds
    - ignore build, dist, and *.egg-info
- Add instructions to DEV_README for how to build and upload a package.
- Update requirements.txt with modules needed to build the package
  (twine and dependencies)
- Create setup.py to describe how to build the package

Guptacos committed:
Set up project to be uploadable to Pypi

- Update gitignore for setuptools builds
    - ignore build, dist, and *.egg-info
- Add instructions to DEV_README for how to build and upload a package.
- Update requirements.txt with modules needed to build the package
  (twine and dependencies)
- Create setup.py to describe how to build the package

spenceryu commented on setup.py:
```diff
@@ -0,0 +1,34 @@
+        'Programming Language :: Python :: 3.7',
+        'Programming Language :: Python :: 3.8',
+    ],
+    # Use ~= to signify that we aren't commiting to Python 4 support yet
```
I thought python 4 isn't released yet, why is ~= on 3.6?

Guptacos replied:
See [here](https://packaging.python.org/guides/distributing-packages-using-setuptools/#python-requires)

Do you think we should just change it to `>=`?

spenceryu commented on requirements.txt:
```diff
@@ -1,10 +1,28 @@
-wheel==0.34.2
 astroid==2.4.1
```
are most of these requires meant for dependencies/build? not too sure on what they actually do.

Guptacos replied:
These are requirements for building the actual spotifython package and for building documentation. It'll only be used during dev

Guptacos committed:
Change ~=3.6 to >=3.6 in setup.py file

No need to lock Python 4 compatibility, we have no idea when Python
4 will be released.
